lib/msf/core/post/file.rb

```diff
@@ -352,31 +352,96 @@ def file_local_write(local_file_name, data)
   #
   # Returns a MD5 checksum of a given remote file
   #
-  # @note THIS DOWNLOADS THE FILE
+  # @note For shell sessions,
+  #       this method downloads the file from the remote host
+  #       unless a hashing utility for use on the remote host is specified.
+  #
   # @param file_name [String] Remote file name
+  # @option util [String] Remote file hashing utility
   # @return [String] Hex digest of file contents
-  def file_remote_digestmd5(file_name)
-    data = read_file(file_name)
-    chksum = nil
-    if data
+  def file_remote_digestmd5(file_name, util: nil)
+    if session.type == 'meterpreter'
+      begin
+        return session.fs.file.md5(file_name)&.unpack('H*').flatten.first
+      rescue StandardError
+        return nil
```

Comment:
Can we get a log here please?

Reply:
Done.

Use of logging / error handling is inconsistent throughout this library (and other Post libs).